### PR TITLE
API provides ability for registered user to bid

### PR DESCRIPTION
**Pull request by Odeane**
### User Story
```
As a registered user
in order to rent a space from the landlord
I would like to be able to bid on listings
```
## Changes proposed in this pull request:
* Registered user can bid

### CODE REVIEW

**Review by Alex-Bellamy — APPROVED** [review]

File: spec/factories/biddings.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
+  factory :bidding do
+    bid { 1.5 }
+    listing_id {65}
```
Comment (oliverochman) [inline spec/factories/biddings.rb]: please refactor to `association :listing`

File: spec/requests/api/v1/bidding/biddings_request_spec.rb [hunk 2]
```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Biddings', type: :request do
+  let!(:user) { create(:user, role: 'registered') }
```
Comment (oliverochman) [inline spec/requests/api/v1/bidding/biddings_request_spec.rb:6]: please correct the indentation in this file

File: app/controllers/api/v1/biddings_controller.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class Api::V1::BiddingsController < ApplicationController
+  before_action :authenticate_user!, only: [:create]
+  def create
+    bidding = current_user.biddings.create(bidding_params)
+    if bidding.persisted?
```
Comment (oliverochman) [inline app/controllers/api/v1/biddings_controller.rb:7]: please correct the indentation in this file

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/models/bidding.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class Bidding < ApplicationRecord
+  validates :bid, presence: true, numericality: { only_integer: true }
+  belongs_to :listing
+  belongs_to :user
```
Comment (oliverochman) [inline app/models/bidding.rb:4]: Add a `before_create` hook here where you check if the current user that is submitting the bid is the same as the landlord of the listing that he wants to bid on. If that is the case raise an error.

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: app/models/bidding.rb (hunk 4 above)
Comment (oliverochman) [reply app/models/bidding.rb:4]: Also add a test for this

**Review by oliverochman — APPROVED** [review]
Good job guys!

**Pull request merged**